engine-comparison/common-harness.sh

```diff
@@ -5,3 +5,47 @@
 # Every "gcloud compute ssh/scp/etc" needs --zone, otherwise throws warning
 GCLOUD_ZONE="us-west1-b"
 
+# Almost definitely fine to be in the public domain
+SERVICE_ACCOUNT="373628893752-compute@developer.gserviceaccount.com"
+
+# Sometimes, gcloud compute instances create returns before a VM is ready to be
+# SSH'ed into. Use this function after "instances create" to prevent skipped calls
+robust_begin_gcloud_ssh () {
+  INSTANCE_NAME=$1
+  while :
+  do
+    ! (gcloud compute ssh $INSTANCE_NAME --command="echo ping"\
+      --zone=$GCLOUD_ZONE 2>&1 | grep "ERROR") && break
+    echo "GCloud VM isn't ready yet. Rerunning SSH momentarily."
+    sleep 5 # arbitrary choice of time
+  done
+}
+
+create_or_start() {
+  INSTANCE_NAME=$1
+  STARTUP_SCRIPT=$2
+
+  if gcloud compute instances describe $INSTANCE_NAME --zone=$GCLOUD_ZONE 2>&1 | grep "ERROR"; then
+    echo "$INSTANCE_NAME doesn't exist yet. Now creating VM."
+    gcloud_create $INSTANCE_NAME $STARTUP_SCRIPT
+  else
+    gcloud compute instances start $INSTANCE_NAME --zone=$GCLOUD_ZONE
+  fi
+
+}
+gcloud_create() {
+  INSTANCE_NAME=$1
+  # If there is a second argument
+  if [[ -n $2 ]]; then
+    STARTUP_SCRIPT_CMD="--metadata-from-file startup-script=$2"
+  fi
+  IMAGE_FAMILY="docker-ubuntu"
```

Comment:
Yeah, there is a level of "customization" which our group will have as constant, but in general should be user-defined